internal/pkg/platform/schema/compiler/extension/primarykey/fixture/expectedSubChildKeyGo.txt

```diff
@@ -0,0 +1,87 @@
+// Code generated by ent extension "primarykey", DO NOT EDIT.
+
+package key
+
+import (
+	"database/sql/driver"
+	"fmt"
+	"github.com/keboola/keboola-as-code/internal/pkg/platform/schema/compiler/extension/primarykey/fixture"
+	"strconv"
+	"strings"
+)
+
+type SubChildKey struct {
+	MyID       fixture.SomeString
+	GroupID    fixture.SomeString
+	CategoryID fixture.SomeInt
+}
+
+// String converts the key to string, it is required for ent validators.
+func (v SubChildKey) String() string {
+	var builder strings.Builder
+	{
+		builder.WriteString("MyID:")
+		builder.WriteString(string(v.MyID))
+	}
+	{
+		builder.WriteString("/")
+		builder.WriteString("GroupID:")
+		builder.WriteString(string(v.GroupID))
+	}
+	{
+		builder.WriteString("/")
+		builder.WriteString("CategoryID:")
+		builder.WriteString(strconv.Itoa(int(v.CategoryID)))
+	}
+	return builder.String()
+}
+
+// Value converts Go struct to database string value.
+func (v SubChildKey) Value() (driver.Value, error) {
+	if err := v.Validate(); err != nil {
+		return nil, err
+	}
+	return v.String(), nil
+}
+
+// Value converts string value from database to Go struct.
+func (v *SubChildKey) Scan(in any) error {
```

Comment:
`Scan` robi deserializaciu `string` -> `struct`.

Reply:
Spomenie, ze vdaka tomu, ze kod je vygenerovany, tak sa nepouziva ziadna `reflection` a je to teda rychle.